workbench-core/user-management/src/plugins/cognitoUserManagementPlugin.ts

```diff
@@ -389,21 +397,28 @@ export class CognitoUserManagementPlugin implements UserManagementPlugin {
   /**
    * Lists the users within the user pool.
    *
-   * @returns an array of {@link User}s
+   * @param request - the request object according to {@link ListUsersRequest}
+   * @returns a {@link ListUsersResponse} object
    *
    * @throws {@link IdpUnavailableError} if Cognito encounters an internal error
    * @throws {@link PluginConfigurationError} if the plugin doesn't have permission to list the users in a user pool
    * @throws {@link PluginConfigurationError} if the user pool id is invalid
    * @throws {@link TooManyRequestsError} if the RPS limit was exceeded
    */
-  public async listUsers(): Promise<User[]> {
+  public async listUsers(request: ListUsersRequest): Promise<ListUsersResponse> {
+    let parsedPaginationToken;
+    if (request.paginationToken) {
+      parsedPaginationToken = fromPaginationToken(request.paginationToken);
+    }
     try {
       const response = await this._aws.clients.cognito.listUsers({
-        UserPoolId: this._userPoolId
+        UserPoolId: this._userPoolId,
+        Limit: request.pageSize,
+        PaginationToken: typeof parsedPaginationToken === 'string' ? parsedPaginationToken : undefined
```

Comment:
Do we need to do this check? When we call `fromPaginationToken`, doesn't it return a string?

Reply:
Btw, are we using `fromPaginationToken` correctly? The [method](https://github.com/aws-solutions/research-service-workbench-on-aws/blob/2645d12a07ae9b9ba185e234713c55639e96291d/workbench-core/base/src/utilities/paginationHelper.ts#L46) returns a dictionary, but we're expecting a string?